h2o-core/src/main/java/water/parser/Parser.java

```diff
@@ -83,30 +89,120 @@ ParseWriter streamParseZip( final InputStream is, final StreamParseWriter dout,
+          headerBytes = ZipUtil.unzipForHeader(din.getChunkData(cidx), this._setup._chunk_size);
+          ParseSetup ps = ParseSetup.guessSetup(null, headerBytes, GUESS_INFO, ParseSetup.GUESS_SEP,
+                  ParseSetup.GUESS_COL_CNT, this._setup._single_quotes, ParseSetup.GUESS_HEADER,
+                  null, null, null, null);
+
+          // check to make sure datasets in file belong to the same dataset
+          // just check for number for number of columns here.  Ignore the column type, user can force it
+          if (this._setup._number_columns == ps._number_columns)
+            goodFile = true;
+          else {
+            String warning = "Your zip file contains a file that belong to another dataset with different " +
+                    "number of column.  Number of columns for files that have been parsed = "+
+                    this._setup._number_columns + ".  Number of columns in new file = "+ps._number_columns+
+                    ".  This new file is skipped and not parsed.";
+            dout.addError(new ParseWriter.ParseErr(warning, -1, -1L, -2L));
+            goodFile = false;
+          }
+
+          // assume column names must appear in the first file.  If column names appear in first and other
+          // files, they will be recognized.  Otherwise, if no column name ever appear in the first file, the other
+          // column names in the other files will not be recognized.
+
+          if (goodFile) {
+            if (ps._check_header == ParseSetup.HAS_HEADER) {
+              if (this._setup._column_names != null) {
+            // found header in later files, only incorporate it if the column names are the same as before
+            String[] thisColumnName = this._setup.getColumnNames();
+            String[] psColumnName = ps.getColumnNames();
+            Boolean sameColumnNames = true;
+            for (int index = 0; index < this._setup._number_columns; index++) {
+              if (!(thisColumnName[index].equals(psColumnName[index]))) {
+                sameColumnNames = false;
+                break;
+              }
+            }
+
+            if (sameColumnNames)
+              this._setup.setCheckHeader(ps._check_header);
+          }
+            } else {  // take care of the case where the last file has header but this file does not.
+              this._setup.setCheckHeader(ps._check_header);
+            }
+          }
+/*          if (this._setup._check_header == ParseSetup.HAS_HEADER) { //check for header on local file
+            this._setup._check_header =
+                    this._setup.parser(_jobKey).fileHasHeader(ZipUtil.unzipForHeader(din.getChunkData(cidx),
+                            this._setup._chunk_size), this._setup);
+          }*/
+        } catch (Exception e) { // something is wrong parsing this file, do not parse in this case
+          String warning = "Your zip file contains a file that we cannot read for some reason";
```

Comment:
It will throw an error when we try to open an input stream on line 238 of ZipUtil.java